Modified views for Q & A Pages

ananyo2012 opened the pull request:
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] tests pass -- `rake test`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [x] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer
* [x] `development.sqlite.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

ananyo2012 commented:
This is the Recently answered view for the questions index page
![recently_answered_v1](https://cloud.githubusercontent.com/assets/10486343/17075236/28c36070-50ab-11e6-813a-4eb046c516a5.png)

ananyo2012 commented:
This is the Asked questions views for user profile page
![asked_question_v1](https://cloud.githubusercontent.com/assets/10486343/17075237/4e5e7a86-50ab-11e6-9e61-68945294fedc.png)

ananyo2012 commented on app/models/drupal_users.rb:
```diff
@@ -66,11 +66,11 @@ def like_count
 
   def liked_notes
-    DrupalNode.includes(:node_selections).where("type = 'note' AND node_selections.liking = true AND node_selections.user_id = ? AND node.status = 1", self.uid).order('node_selections.nid DESC')
+    DrupalNode.includes(:node_selections).where("type = 'note' AND node_selections.liking = ? AND node_selections.user_id = ? AND node.status = 1", true, self.uid).order('node_selections.nid DESC')
```
Had to make this change because the sqlite db was showing an error probably because it wasn't taking true as a boolean.

ananyo2012 commented on app/views/comments/index.html.erb:
```diff
@@ -1,26 +1 @@
-</table>
-
-<%= will_paginate @comments, :renderer => BootstrapPagination::Rails %>
+<%= render partial: 'comments/comments' %>
```
@jywarren Moved the comments view to a separate partial as the comments action in the user controller was previously using the `home/comments` partial which was broken after the previous changes you made. Since rendering the `comments/index` template also included the header and footer so the only option was to make a separate template for it.

ananyo2012 commented on app/views/questions/_questions.html.erb:
```diff
@@ -18,7 +18,7 @@
         </span>
         <p style="margin-top:30px;"><% if logged_in_as(['admin','moderator']) %><a class="btn btn-default btn-xs" href="/moderate/spam/<%= node.id %>"><i class="fa fa-ban"></i> Spam</a><% end %>
-        <a href="#" class="btn btn-default btn-xs">Post an answer</a></p>
+        <a href="<%= node.path(:question) %>#post-answer" class="btn btn-default btn-xs">Post an answer</a></p>
```
Wierd! the Post an answer link that I added was somehow reverted in any previous changes. I added it back. Nevertheless it previously wan't correct so I put the correct one this time.

ananyo2012 commented on app/controllers/questions_search_controller.rb:
```diff
@@ -37,11 +37,11 @@ def typeahead
-      link = "<span class='pull-left'><i data-url='" + match.path(:question) + 
-             "' class='fa fa-question-circle'></i> " + match.title + "</span>"
+      link = "<i data-url='" + match.path(:question) + 
+             "' class='fa fa-question-circle'></i> " + match.title
```
This fixes the problem of dropdown link in the questions search box which is not working currently.

ananyo2012 commented on app/controllers/users_controller.rb:
```diff
@@ -115,6 +115,15 @@ def profile
                        .where(type: 'note', status: 1, uid: @user.uid)
+                       .joins(:drupal_tag)
+                       .where('term_data.name NOT LIKE ?', 'question:%')
+                       .group('node.nid')
```
@jywarren This query is not working as expected. I wan't to exclude any questions from the research notes section. But some of the questions are still included with this query. Some help regarding this would be great.

jywarren replied:
Ah, i see. I believe grouping by node.nid ought to then show only one record per node, and so this looks OK in that respect. But are you joining the right table?  It also seems right; see: https://github.com/publiclab/plots2/blob/master/app/controllers/tag_controller.rb#L9

Let me think. If it's inconsistent, perhaps it works when the node has only a single tag and it's `question:%` but if it has other tags, then the  `question:%` is excluded but they get included anyways due to the other tags. I've seen this before.

So you need two queries, I believe. One which collects the `nids` to exclude, and a second which queries with those `nids` excluded. A similar question came up in this discussion: https://groups.google.com/forum/#!searchin/plots-dev/exclude/plots-dev/HUfgxx5kwPA/Z3pQhR1fDwAJ

The "EXCEPT" query should've worked, or we could have looked further into using a subquery, but for expediency, I just wrote simpler code that was not quite as efficient: https://github.com/publiclab/plots2/issues/375

Consider that in my `NOT IN` code, you are holding the total # of nids that you'd like to exclude in memory, so try to narrow your first query well so as not to collect ALL nids every time.

ananyo2012 replied:
@jywarren Thanks for the help. It is working as expected now.

ananyo2012 commented:
Also includes commits from #636.

jywarren commented:
what if:

1. Answered questions were clearly marked as Answered (maybe with the same green indicator used on the Answer?)

2. Profiles just had an "Questions" tab, then two sections below for Answered and Asked?

ananyo2012 commented:
Yeah it's a good idea Jeff. I will work on it. I was thinking of the
listing the answers in the profile page too. So should we list the quoted
answer somewhere near the questions for the Answered section?
ᐧ

On Thu, Jul 28, 2016 at 10:09 PM, Jeffrey Warren <notifications@github.com>
wrote:

> what if:
>
>    1.
>
>    Answered questions were clearly marked as Answered (maybe with the
>    same green indicator used on the Answer?)
>    2.
>
>    Profiles just had an "Questions" tab, then two sections below for
>    Answered and Asked?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/628#issuecomment-235952048>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKACR2fApXECw4dCcz1ALjQ5YFxJhJifks5qaNs0gaJpZM4JTQnt>
> .
>

jywarren commented:
Hmm, that would be a lot to show; maybe we could just say "Answered by
*username*" or "6 proposed answers" ?

On Thu, Jul 28, 2016 at 12:42 PM, Ananya Maiti <notifications@github.com>
wrote:

> Yeah it's a good idea Jeff. I will work on it. I was thinking of the
> listing the answers in the profile page too. So should we list the quoted
> answer somewhere near the questions for the Answered section?
> ᐧ
>
> On Thu, Jul 28, 2016 at 10:09 PM, Jeffrey Warren <notifications@github.com
> >
> wrote:
>
> > what if:
> >
> > 1.
> >
> > Answered questions were clearly marked as Answered (maybe with the
> > same green indicator used on the Answer?)
> > 2.
> >
> > Profiles just had an "Questions" tab, then two sections below for
> > Answered and Asked?
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/publiclab/plots2/pull/628#issuecomment-235952048>,
> or mute
> > the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AKACR2fApXECw4dCcz1ALjQ5YFxJhJifks5qaNs0gaJpZM4JTQnt
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/628#issuecomment-235953029>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJyF2F92HIfHAA_YQSGaS9F7DQcfBks5qaNwMgaJpZM4JTQnt>
> .
>

ananyo2012 commented:
Yes that's a better idea; just linking the answer with a text.
ᐧ

On Thu, Jul 28, 2016 at 10:16 PM, Jeffrey Warren <notifications@github.com>
wrote:

> Hmm, that would be a lot to show; maybe we could just say "Answered by
> *username*" or "6 proposed answers" ?
>
> On Thu, Jul 28, 2016 at 12:42 PM, Ananya Maiti <notifications@github.com>
> wrote:
>
> > Yeah it's a good idea Jeff. I will work on it. I was thinking of the
> > listing the answers in the profile page too. So should we list the quoted
> > answer somewhere near the questions for the Answered section?
> > ᐧ
> >
> > On Thu, Jul 28, 2016 at 10:09 PM, Jeffrey Warren <
> notifications@github.com
> > >
> > wrote:
> >
> > > what if:
> > >
> > > 1.
> > >
> > > Answered questions were clearly marked as Answered (maybe with the
> > > same green indicator used on the Answer?)
> > > 2.
> > >
> > > Profiles just had an "Questions" tab, then two sections below for
> > > Answered and Asked?
> > >
> > > —
> > > You are receiving this because you authored the thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/publiclab/plots2/pull/628#issuecomment-235952048>,
> > or mute
> > > the thread
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AKACR2fApXECw4dCcz1ALjQ5YFxJhJifks5qaNs0gaJpZM4JTQnt
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/publiclab/plots2/pull/628#issuecomment-235953029>,
> or mute
> > the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AABfJyF2F92HIfHAA_YQSGaS9F7DQcfBks5qaNwMgaJpZM4JTQnt
> >
> > .
>
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/628#issuecomment-235954108>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKACRw9IXqNtYR_9E9DpahAmuPdeFIkQks5qaNz_gaJpZM4JTQnt>
> .
>

jywarren commented:
Shall I merge this now, or do you want to keep working on it? (or in another PR later?)

ananyo2012 commented:
@jywarren Please wait till I complete all the features.

ananyo2012 commented:
Probably it will be ready for merge by Tuesday.

jywarren commented:
Perfect, no problem.

On Jul 29, 2016 2:34 PM, "Ananya Maiti" <notifications@github.com> wrote:

> Probably it will be ready for merge by Tuesday.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/628#issuecomment-236259187>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ4m81rQiPLA9MrsT9jbMt3HWT2-Nks5qakelgaJpZM4JTQnt>
> .
>

ananyo2012 commented:
@jywarren Just some clarifications on what you asked
`Answered questions were clearly marked as Answered (maybe with the same green indicator used on the Answer?)`
When will a question marked as answered? When there is atleast one answer or when it is marked as accepted by the question author?

jywarren commented:
When an answer is accepted. Thanks!

ananyo2012 commented:
@jywarren This is almost complete. Just need to add some more tests. You can review it in the mean time. I am posting the screenshots for the modified views below.

ananyo2012 commented:
I have already posted a screenshot of the recently answered view. Link to the question page is added in the navbar under the Projects tab. The question links in dashboard now link to the new question views. Also the Ask question links in the Dashborad and tags page link to the question post page now.

ananyo2012 commented:
I have added redirect  links to the questions page wherever applicable so that question nodes redirect to the question url and there are no duplicate versions of question nodes.

ananyo2012 commented:
I have introduced two new methods in the drupal node model for finding research notes and questions. Notes pages now only list research notes and question page only list questions. This definition follows in other places where research notes and question list separately like in the tags page and user profile page.

ananyo2012 commented:
This how the questions index page looks now
![answer_label](https://cloud.githubusercontent.com/assets/10486343/17454967/0e2d3ff2-5bc7-11e6-840f-5d9db04a2d01.png)

ananyo2012 commented:
Answered questions are labeled with a green answered tag as shown. You can also see a sort by tags field in the sidebar. You can sort the question based on tags by entering the question tag name. It supports the autocomplete feature like other tag input. 

Basically it stores the tagname in session in a hash with the keys being the tag ids. Heres a closer look  on how it looks when you sort it by tagnames.
![sort_by_tags](https://cloud.githubusercontent.com/assets/10486343/17455012/8777cb60-5bc8-11e6-8e69-98a19b210a21.png)
You can clear all tags by clicking the Clear all tags link that is present just below all tagnames. Also you can individually remove any tags from the sort.

ananyo2012 commented:
This is the view of the Asked section of the Questions tab in user profile page. It shows all questions asked by the user.
![profile_asked](https://cloud.githubusercontent.com/assets/10486343/17456797/40bd117e-5c01-11e6-8cd2-b4fab4a9984d.png)

ananyo2012 commented:
This is the answered section of the user profile page.
![profile_answered_1 0](https://cloud.githubusercontent.com/assets/10486343/17456849/913060ec-5c02-11e6-9ed0-044819c77f97.png)
On hovering over the dropdown button a tooltip is shown with the no of answers given by the user to that question. If any one of the user's answer in accepted then an accepted label is shown beside the question title.
![profile_answered_1 2](https://cloud.githubusercontent.com/assets/10486343/17456864/298c3776-5c03-11e6-88e0-292f63b9ccd9.png)

ananyo2012 commented:
On clicking the dropdown button the answer links to the question which the user answered are shown along with the date on the which the answer was posted(the link is given in the dates). The no of likes for the answer is shown beside the answer link. A green checkbox is shown beside the answer which is accepted.
![profile_answered_1 1](https://cloud.githubusercontent.com/assets/10486343/17456884/db2d7e9a-5c03-11e6-9778-a276a4107671.png)

ananyo2012 commented:
Now moving to the sidebar of the question show page. The sidebar contains a **Ask question** and an **Ask related question** button. I have removed the Ask question button from the top of the question title as it was looking quite odd there. They look better in the sidebar. The sidebar lists the top answerers to that question. A tooltip with the no of answers is shown on hovering over the user profile links. Also related questions tagged with the same tags is shown in the sidebar.
![question_sidebar](https://cloud.githubusercontent.com/assets/10486343/17456911/d17abae2-5c04-11e6-8480-af155c710648.png)

ananyo2012 commented:
The tags pages now have a Questions section where all questions tagged with that tag are listed. The research notes section only list the research notes tagged with that tag.
![question_tags_1 0](https://cloud.githubusercontent.com/assets/10486343/17456936/a737fb4a-5c05-11e6-9399-a6975c26acff.png)

ananyo2012 commented:
In case of question power tags there is no research section since by default all question power tags point to a question.
![question_tags_1 1](https://cloud.githubusercontent.com/assets/10486343/17456953/f4736fb6-5c05-11e6-8a9c-5d2ee882715a.png)
So another change introduced with this is all question tags by default have a question type and link to the Question section. And tags other than question tags by default link to the Research note section.

ananyo2012 commented:
@steviepubliclab I have linked all the tags in the question views to the Question tags page. In this way we can see all questions tagged with that tag and don't need to do any search.

ananyo2012 commented:
Finally some work on the question search functionality. Previously a redirect was made to the Question post page in case no search results were found which was a little baffling for a first time user. So I added a Proceed to question button iin the question search views. This is how it looks when a seacrh result is found.
![questions_search_v1 0](https://cloud.githubusercontent.com/assets/10486343/17457009/c05f927a-5c07-11e6-9a9b-4c8b57ca2b93.png)
When no search results are found it also clearly mentions that and then user can Proceed to their question titled with the search term.
![questions_search_v1 1](https://cloud.githubusercontent.com/assets/10486343/17457023/fa2eeb68-5c07-11e6-8136-b9d6c5dac784.png)

ananyo2012 commented:
Also I did some modifications in the Question Search field and Ask button. Previously the search box and the ask button was Submitted to the same url. But now pressing the Ask button takes you the question post page with the title entered in the search box. The search form is submitted via a get request. The tooltip text clearly mentions that. So in case the user directly wants to move to the question post page he can press the Ask button after mentioning the title. 

While the search input on submit shows the search results. It is submitted via a javascript submit. Also autocomplete results are shown as you type in. The input box only shows a "Search a question" placeholder text now.
![question_ask_button](https://cloud.githubusercontent.com/assets/10486343/17457045/156e785c-5c09-11e6-965a-1d44a3754695.png)

ananyo2012 commented:
@jywarren Added tests for the changes. This is now ready. Please review.

jywarren commented:
Oh wow, so many changes to look at. It looks great, thanks -- going through one by one:

* Could "Sort by tags" be renamed "Refine by tag"
* Could "Refine by tag" be added just below the tabs as the [ Asked ] [ Answered ] buttons are? I know this is difficult to fit, but I think it's important that the refining be strongly related to the results, and not in the sidebar, for clarity.
* For "On hovering over the dropdown button a tooltip is shown with the no of answers given by the user to that question." -- does that mean the # of answers by the same user? I don't imagine a user giving more than one answer, or do you mean all answers to that question? 
* for the above, could the "dropdown" circular button be made to be a button like the `Post an answer` button in the Question template -- perhaps with a down-caret after the text? And could the header of the box still show a `(?)` icon -- esp. since the question title is still shown?
* Can the drop-down shown answers be on a white background with 3px more padding so they're spaced out a bit more?
* a space is needed after the icon in "Top Answerers"
* can we center "Proceed to your question" button?

That's all I can see for now -- this is a really exciting update, thanks!

ananyo2012 commented:
@jywarren I am working now on the changes you mentioned. For the # 3 mentioned above The tooltip text on hovering over the dropdown button it shows the no of answers posted by the user. A link below it shows total no of answers. Yes it is unlikely for an user to post multiple answers but just in case they do so not ruling out the possibility. What do you think?

ananyo2012 commented:
@jywarren Could we change the title to "Refine Question by tags". Putting it below the tabs really looks odd. It best fits in the sidebar.

ananyo2012 commented:
@jywarren I have made the changes to the user answered views as you said .Here are the updated screenshots.
![profile_answered_v2 1](https://cloud.githubusercontent.com/assets/10486343/17545969/68d6c4d0-5efd-11e6-81cf-ccb0a2efb333.png)
![profile_answered_v2 2](https://cloud.githubusercontent.com/assets/10486343/17546102/1d5123d8-5efe-11e6-8e5b-57a176af597a.png)

ananyo2012 commented:
I also centered the "Proceed to question" button. Here is the screenshot.
![question_search_v2 1](https://cloud.githubusercontent.com/assets/10486343/17546264/ef457614-5efe-11e6-9b8c-2f950e95fb45.png)
Also added space after the user icon in Top Answerers section.

jywarren commented:
"Refine questions by tag" maybe is best, and I agree it's a tough design choice -- let's go with sidebar for now and see how it works. 

> the dropdown button shows the no of answers posted by the user

That sounds good!

I think the caret for the dropdown should be this: http://fontawesome.io/icon/caret-down/

And it really looks more correct in a button of the same type as the [Post an Answer] buttons, like:

`# answers v ` (where `v` is the caret). Make sense? And once they press that, "All # answers" maybe is unnecessary -- they've already pressed once? Just show them all, perhaps?

Thanks!

ananyo2012 commented:
@jywarren Actually the point of showing the Answered section in the user profile page is to show just the answers posted by the user to the question. So we should list only the links to the answers the user posted in the answered section. The `Show all # answers` below all the answer links(posted only by that user) is an additional button that links to the all answers section of that question.

ananyo2012 commented:
@jywarren This is how it looks now.
![profile_answered_v2 3](https://cloud.githubusercontent.com/assets/10486343/17560799/fd28921a-5f40-11e6-8cc0-97295b61df6c.png)

jywarren commented:
Oh, i see -- then maybe we just need to say "Your answers" -- although really we'll have to see how it looks when real Q&As are used... maybe there will seldom be enough total answers that we need to separate them out. We'll have to see. 

But "All answers" could just be a link instead of a button. 

Can the "caret" button be styled just like this?

![screenshot 2016-08-12 at 10 33 51 am](https://cloud.githubusercontent.com/assets/24359/17618549/4f15cff6-6078-11e6-865d-5645595b8bb7.png)

And say `Your answers v `?

I think we're almost there on this. Happy to do a merge over the weekend if I can find time (i'll by flying).

ananyo2012 commented:
@jywarren How about this Jeff? Is it all right now?
![profile_answered_v3 1](https://cloud.githubusercontent.com/assets/10486343/17628977/89a3276a-60d5-11e6-8090-1ecca036e280.png)
The text in the dropdown is "Answers by username". This is more appropriate as the profile is visible to everyone. Also I have removed the tooltip from the button since the username is shown in the button now. And since the no of answers is significantly going to be just one most of the time so it is better to just dropout the no of answers. Also you can notice the "Show all # answers" is simple a link now.

jywarren commented:
Hi, Ananyo -- I tried posting a question, then answering it myself, and don't see it listed under Asked or Answered on my profile. Can you try writing a functional test for this to see what's up?

jywarren commented:
Also, I think I prefer the button be colored the same as the one I referenced, plus not overhanging the bottom edge of the box. Finally, I think it may look better for the `Answered` box to be green outlined with green text, but transparent background interior. 

Please forgive my detailed requests on design issues; I'm just really excited for this to publish and I want it to look as fantastic as possible. Thanks!!

ananyo2012 commented:
@jywarren Uh I see everything Ok at my end. I also added a test previously. I added two more assert_select lines to further confirm it. See from this line: https://github.com/publiclab/plots2/pull/628/files#diff-a3e88deed90ce691176678568ded7007R229. Just to be sure I have updated my commits. Did you pull in the latest commits?

ananyo2012 commented:
@jywarren By the Answered box do you mean the white area where all the answer links are shown?

jywarren commented:
I'll try again. By box I mean button, thats centered and white, with the
caret -- lets have it not overhang, just be inside, as in the example
screenshot I posted above.

Thanks!

On Aug 14, 2016 2:03 PM, "Ananya Maiti" <notifications@github.com> wrote:

> @jywarren <https://github.com/jywarren> By the Answered box do you mean
> the white area where all the answer links are shown?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/628#issuecomment-239675260>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ_rv3JI30cQO5MyHrfXLO7-oQWbJks5qfyA7gaJpZM4JTQnt>
> .
>

jywarren commented:
I found the issue -- i had not approved the question that had been posted. But when I press approve on it, I'm actually getting this error:

```
undefined method `time_ago_in_words' for #<AdminController:0x3273120>
app/controllers/admin_controller.rb:119:in `publish'
```

That's within 

```ruby
#{time_ago_in_words(@node.created_at)}
```

That code [hasn't been touched for months](https://github.com/publiclab/plots2/blame/master/app/controllers/admin_controller.rb), but I wonder if it's a `friendly_id` related issue? Can you take a look? 

Also, I don't see a "this post is in moderation" message on the top of a question as I would on the top of a note. Can we add a test for the above and for this issue?

jywarren commented:
Oops, scrap the 2nd issue above -- it had approved the post, so that's why the notice at the top didn't appear. If there's not already a test for that, it's probably a good idea, though.

jywarren commented:
And `http://localhost:3000/profile/admin` now shows the question and answer correctly, thanks. Can we put `Answered` in the answer view onto its own line? It's inheriting the font style of the title, which is the wrong font. 

OK, pending the `time_ago_in_words` error and the other style tweaks above, this is almost there. Thanks!

jywarren commented:
It doesn't look like we have a test for admin (as opposed to a moderator) approving a first-timer post. I've added one and it seems to pass, so I'm not sure what's up. Maybe we should add one for approving a first-timer question, not a normal note.

ananyo2012 commented:
@jywarren Updated and rebased on the latest master. Here is the screenshot
![profile_answered_v3 2](https://cloud.githubusercontent.com/assets/10486343/17649891/024a99b8-625d-11e6-9da0-52e646a88f5b.png)

jywarren commented:
Oh, i'm sorry, i meant green outlining of the `Accepted` badge, not the `answers by user`. `Answers by user looks perfect, just needs the same color as the (?) icon -- not light green. `Accepted` could be green borders with transparent interior, and on its own line. 

Looking great on spacing, though!

ananyo2012 commented:
@jywarren Ok updating it. I am also trying to look at the issue you mentioned. It looks like I have missed something for the first timer questions.

jywarren commented:
Ok, good luck! I am about to board a plane so won't be checking in again
today. Thanks as usual!

On Aug 14, 2016 3:00 PM, "Ananya Maiti" <notifications@github.com> wrote:

> @jywarren <https://github.com/jywarren> Ok updating it. I am also trying
> to look at the issue you mentioned. It looks like I have missed something
> for the first timer questions.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/628#issuecomment-239678107>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ_vxtUCieVxoCaQ2Zxu3lPrfxmG2ks5qfy2PgaJpZM4JTQnt>
> .
>

ananyo2012 commented:
@jywarren Yes it is really looking great this way now.
![profile_answered_v3 3](https://cloud.githubusercontent.com/assets/10486343/17650069/4dd47422-6261-11e6-8655-b100623a9eff.png)

ananyo2012 commented:
@jywarren First timer questions are now visible to admins in the questions index page. I have also modified the tests to justify it. Also flash messages are displayed while approving the first timer questions through proper redirects. Also added test for admin controller to support it.

ananyo2012 commented:
I have also changed the Answered label in the questions view similar to that in the profile answered views.
![profile_asked_v2 1](https://cloud.githubusercontent.com/assets/10486343/17652176/5df9974c-6293-11e6-9094-87e6cc3225de.png)

jywarren commented:
Great, i'll do final review and merge tomorrow. Great work!

On Sun, Aug 14, 2016 at 10:23 PM, Ananya Maiti <notifications@github.com>
wrote:

> I have also changed the Answered label in the questions view similar to
> that in the profile answered views.
> [image: profile_asked_v2 1]
> <https://cloud.githubusercontent.com/assets/10486343/17652176/5df9974c-6293-11e6-9094-87e6cc3225de.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/628#issuecomment-239698653>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ-HYo0Bq05zSmCRzDkKajLwy38iJks5qf4c4gaJpZM4JTQnt>
> .
>

jywarren commented:
💯 merged!! I made some small final style tweaks, and hope to publish tonight. GREAT WORK, ananyo! Reach out to Liz and Stevie and ask them to help you test it out once it's published. I'll alert you.

ananyo2012 commented:
Thanks Jeff! I was thining of adding some documentation for Q & A in the
same PR. Anyways I will make a new one for it.
ᐧ

On Wed, Aug 17, 2016 at 3:42 AM, Jeffrey Warren <notifications@github.com>
wrote:

> 💯 merged!! I made some small final style tweaks, and hope to publish
> tonight. GREAT WORK, ananyo! Reach out to Liz and Stevie and ask them to
> help you test it out once it's published. I'll alert you.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/628#issuecomment-240256089>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKACR4_hAx8FF2aqtvL7SndIPgQv8r_Dks5qgjXjgaJpZM4JTQnt>
> .
>
